### ISSUE TITLE
Are there any plans to add support for using `evalsha` with `RLock`?

### ISSUE THREAD

**Issue by zhumengzhu** [redisson/redisson#5434]
hi @mrniko, after some investigation, it appears that the problem labeled as #3656 may not have been resolved(by v3.24.3). 

It seems fixing this is not easy; could you identify the challenges?

We've noticed that, occasionally, the eval command takes more than 200 milliseconds to execute in our production environment:
<img width="1471" alt="image" src="https://github.com/redisson/redisson/assets/4508623/9fd2daa9-af1d-4d35-8453-d23366957c70">

Here is a test case to demonstrate:
```Java
import org.redisson.Redisson;
import org.redisson.api.RLock;
import org.redisson.api.RedissonClient;
import org.redisson.config.Config;

import java.time.LocalDateTime;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;
import java.util.concurrent.ThreadLocalRandom;
import java.util.concurrent.TimeUnit;

public class RedissonLockTest {

    public static void main(String[] args) {
        // Init
        Config config = new Config();
        config.useSingleServer().setAddress("redis://127.0.0.1:6379");
        RedissonClient client = Redisson.create(config);

        final int nThreads = 4;
        ExecutorService pool = Executors.newFixedThreadPool(nThreads);
        for (int i = 0; i < nThreads; i++) {
            pool.submit(() -> {
                String threadName = Thread.currentThread().getName();
                for (int j = 0; j < Integer.MAX_VALUE; j++) {
                    String lockName = threadName + ":" + j;
                    RLock lock = client.getLock(lockName);
                    // RLock lock = client.getFairLock(lockName);
                    try {
                        //
                        long lockStartTime = System.currentTimeMillis();
                        if (lock.tryLock(5, 5, TimeUnit.SECONDS)) {
                            long lockTime = System.currentTimeMillis() - lockStartTime;
                            if (lockTime > 10) {
                                System.err.printf("@@@@!!!! tryLock. lockName=%-25s time=%-30s cost=%sms%n", lockName, LocalDateTime.now(), lockTime);
                            }

                            try {
                                // sleep for random time
                                long sleepTime = ThreadLocalRandom.current().nextLong(30, 70);
                                TimeUnit.MILLISECONDS.sleep(sleepTime);
                            } finally {
                                long unlockStartTime = System.currentTimeMillis();
                                lock.unlock();
                                long unlockTime = System.currentTimeMillis() - unlockStartTime;
                                if (unlockTime > 10) {
                                    System.err.printf("@@@@!!!!  unlock. lockName=%-25s time=%-30s cost=%sms%n", lockName, LocalDateTime.now(), unlockTime);
                                }
                            }
                        }
                    } catch (InterruptedException e) {
                        e.printStackTrace();
                    }
                }
            });
        }
    }

}

```

> result
@@@@!!!! tryLock. lockName=pool-1-thread-3:0         time=2023-11-16T18:19:54.028        cost=14ms
@@@@!!!! tryLock. lockName=pool-1-thread-1:0         time=2023-11-16T18:19:54.028        cost=14ms
@@@@!!!! tryLock. lockName=pool-1-thread-4:0         time=2023-11-16T18:19:54.028        cost=14ms
@@@@!!!! tryLock. lockName=pool-1-thread-2:0         time=2023-11-16T18:19:54.028        cost=15ms
@@@@!!!! tryLock. lockName=pool-1-thread-4:11        time=2023-11-16T18:19:54.683        cost=40ms
@@@@!!!! tryLock. lockName=pool-1-thread-2:12        time=2023-11-16T18:19:54.683        cost=40ms
@@@@!!!!  unlock. lockName=pool-1-thread-1:415       time=2023-11-16T18:20:16.148        cost=13ms
@@@@!!!!  unlock. lockName=pool-1-thread-2:405       time=2023-11-16T18:20:16.148        cost=13ms
@@@@!!!!  unlock. lockName=pool-1-thread-4:665       time=2023-11-16T18:20:30.657        cost=14ms
@@@@!!!! tryLock. lockName=pool-1-thread-1:688       time=2023-11-16T18:20:30.657        cost=17ms
@@@@!!!!  unlock. lockName=pool-1-thread-3:680       time=2023-11-16T18:20:30.918        cost=12ms
@@@@!!!!  unlock. lockName=pool-1-thread-1:691       time=2023-11-16T18:20:30.918        cost=25ms

**Comment by mrniko** [comment]
Usage of evalsha command doesn't mean that the commands start to execute faster. No words about that in docs. Evalsha is applied for RLock. https://github.com/redisson/redisson/issues/3656 was resolved.

**Comment by zhumengzhu** [comment]
> Usage of evalsha command doesn't mean that the commands start to execute faster. No words about that in docs. Evalsha is applied for RLock. #3656 was resolved.

While #3656 may have been resolved(), when I debugged it with v3.24.3, I found that it still indeed uses `eval` instead of `evalsha`, any idea?
<img width="1444" alt="Pasted Graphic 4" src="https://github.com/redisson/redisson/assets/4508623/ab1f9e9e-b35a-4402-bb68-14dc8a06d03b">
